File: Swift-C++-Testsuite/test/basicOperators/swift/ComparisonNumber.swift
@main
struct M42Main {

    static func main() {
        var _ = numberEqual(x: 42)  // true
        var _ = numberEqual(x: 41)  // false
        var _ = numberNotEqual(x: 42)  // false
        var _ = numberNotEqual(x: 41)  // true
        var _ = numberLarger(x: 43)  // true
        var _ = numberLarger(x: 41)  //false
        var _ = numberSmaller(x: 41)  // true
        var _ = numberSmaller(x: 43)  // false
        var _ = numberLargerOrEqual(x: 42)  // true
        var _ = numberLargerOrEqual(x: 41)  // false
        var _ = numberSmallerOrEqual(x: 42)  // true
        var _ = numberSmallerOrEqual(x: 43)  // false
    }

    static func numberEqual(x: Int) -> Bool {
        var tmp = x == 42
        return tmp
    }
    static func numberNotEqual(x: Int) -> Bool {
        var tmp = x != 42
        return tmp
    }
    static func numberLarger(x: Int) -> Bool {
        var tmp = x > 42
        return tmp
    }
    static func numberSmaller(x: Int) -> Bool {
        var tmp = x < 42
        return tmp
    }
    static func numberLargerOrEqual(x: Int) -> Bool {
        var tmp = x >= 42
        return tmp
    }
    static func numberSmallerOrEqual(x: Int) -> Bool {
        var tmp = x <= 42
        return tmp
    }
}

File: Swift-C++-Testsuite/test/basicOperators/C++/ComparisonNumber.cpp

bool numberEqual(int x) {
    bool tmp = x == 42;
    return tmp;
}

bool numberNotEqual(int x) {
    bool tmp = x != 42;
    return tmp;
}
bool numberLarger(int x) {
    bool tmp = x > 42;
    return tmp;
}
bool numberSmaller(int x) {
    bool tmp = x < 42;
    return tmp;
}
bool numberLargerOrEqual(int x) {
    bool tmp = x >= 42;
    return tmp;
}
bool numberSmallerOrEqual(int x) {
    bool tmp = x <= 42;
    return tmp;
}

int main() {
    bool a1 = numberEqual(42);
    bool a2 = numberEqual(41);
    bool b1 = numberNotEqual(42); // false
    bool b2 = numberNotEqual(41); // true
    bool c1 = numberLarger(43); // true
    bool c2 = numberLarger(41); //false
    bool d1 = numberSmaller(41); // true
    bool d2 = numberSmaller(43); // false
    bool e1 = numberLargerOrEqual(42); // true
    bool e2 = numberLargerOrEqual(41); // false
    bool f1 = numberSmallerOrEqual(42); // true
    bool f2 = numberSmallerOrEqual(43); // false
}
